### PR TITLE
Phy tests

### PR DESCRIPTION
**Pull request by kenr**
* Add test for changing physical interface
* Add textual description of remaining SDv6 GAP events

### CODE REVIEW

File: test/softdevice_api/test_phy_update.cpp [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+// The test framework is not thread safe so this variable is used to communicate that an issues has
+// occurred in a callback.
+bool error = false;
+
+// Set to true when the test is complete
+bool testComplete = false;
+
+using namespace testutil;
+
+TEST_CASE("test_phy_update")
```
Comment (bihanssen) [inline test/softdevice_api/test_phy_update.cpp]: Great to see phy updates covered as a test case.

File: test/include/test_util_conversion.h [hunk 2]
```diff
@@ -626,17 +626,114 @@ static std::string asText(const ble_gap_adv_report_type_t &reportType)
...
 static std::string asText(const ble_gap_aux_pointer_t &auxPointer)
 {
     std::stringstream retval;
     retval << "aux_offset:" << static_cast<uint32_t>(auxPointer.aux_offset);
     retval << " aux_phy:" << static_cast<uint32_t>(auxPointer.aux_phy);
     return retval.str();
 }
-#endif
+
+static std::string asText(const ble_gap_phys_t &phys)
```
Comment (bihanssen) [inline test/include/test_util_conversion.h]: Are these event-to-text conversions intended to be used only in test context? Could be useful to include in log file also for normal operation.

Comment (kenr) [reply test/include/test_util_conversion.h]: @bihanssen, currently they are only in the tests. Adding them in pc-ble-driver log framework is something we could do. Perhaps a better solution is to add them in a separate C++ library that encapsulates the pc-ble-driver.

Comment (bihanssen) [reply test/include/test_util_conversion.h]: Ok, something for the backlog then, Good logging is certainly valuable to have.

**Pull request merged**